core/src/main/scala/org/apache/spark/ui/WebUI.scala

```diff
@@ -134,7 +134,8 @@ private[spark] abstract class WebUI(
   def bind() {
     assert(!serverInfo.isDefined, "Attempted to bind %s more than once!".format(className))
     try {
-      serverInfo = Some(startJettyServer("0.0.0.0", port, sslOptions, handlers, conf, name))
+      var host = Option(conf.getenv("SPARK_LOCAL_IP")).getOrElse("0.0.0.0")
+      serverInfo = Some(startJettyServer(host, port, sslOptions, handlers, conf, name))
       logInfo("Started %s at http://%s:%d".format(className, publicHostName, boundPort))
```

Comment:
I am in an assumption that we need to consider SPARK_PUBLIC_DNS value for showing the url in the log as per our previous conversation. Don’t we need to consider SPARK_PUBLIC_DNS while logging here?